src/openvic-simulation/economy/Producer.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+	class Producer {
+	public:
+		ProductionType const& PROPERTY(production_type);
+	};
+
+	class Manufacturer : public virtual Producer {
+	public:
+		Good::good_map_t PROPERTY(stockpile);
+	};
+
+	class ArtisanalProducer final : public Manufacturer {
+	public:
+		fixed_point_t PROPERTY(current_production);
+		Good::good_map_t PROPERTY(current_needs);
+	};
+
+	class NonArtisanalProducer : public virtual Producer {
+	public:
+		fixed_point_t PROPERTY(revenue_yesterday);
+		fixed_point_t PROPERTY(output_quantity_yesterday);
+		fixed_point_t PROPERTY(unsold_quantity_yesterday);
+		fixed_point_t PROPERTY(size_multiplier);
+		ordered_map<Pop const*, Pop::pop_size_t> PROPERTY(employees);
+	};
+
+	class FactoryProducer final : public NonArtisanalProducer, public Manufacturer {
+	private:
+		static constexpr size_t days_of_history = 7;
+		int PROPERTY(profit_history_current);
+		std::array<fixed_point_t, days_of_history> PROPERTY(daily_profit_history);
+
+	public:
+		fixed_point_t PROPERTY(budget);
+		fixed_point_t PROPERTY(balance_yesterday);
+		fixed_point_t PROPERTY(received_investments_yesterday);
+		fixed_point_t PROPERTY(market_spendings_yesterday);
+		fixed_point_t PROPERTY(paychecks_yesterday);
+		uint32_t PROPERTY(unprofitable_days);
+		uint32_t PROPERTY(injected_days);
+		uint32_t PROPERTY(days_without_input);
+		uint8_t PROPERTY_RW_ACCESS(hiring_priority, public);
```

Comment:
Why bother with the property stuff if they are just private fields anyway?
I made them properties as I want to expose the getter as public members.

Reply:
`PROPERTY` does create public getters, then it switches back to private so that the variables themselves aren't public, which would defeat the purpose of the getters. `PROPERTY_RW` also makes the setter public.